src/library/scanner/recursivescandirectorytask.cpp

```diff
@@ -79,14 +79,11 @@ void RecursiveScanDirectoryTask::run() {
     if (prevHash != newHash) {
         // Rescan that mofo! If importing fails then the scan was cancelled so
         // we return immediately.
-        if (!filesToImport.isEmpty()) {
```

Comment:
Instead, could you keep the ```emit(directoryHashed(...))``` for the empty case? That would prevent scheduling tons of no-op tasks if the user has a bunch of directories with no supported files for some reason.